apps/lend/src/entities/token/lib/validation.ts

```diff
@@ -0,0 +1,15 @@
+import { CombinedTokenParams, TokenQueryKeyType } from '@/entities/token'
+import { assertValidity, checkValidity } from '@/shared/lib/validation'
+import { tokenValidationSuite } from '@/entities/token/model/validation'
+
+export const assertTokenValidity = <T extends CombinedTokenParams>(data: T, fields?: Extract<keyof T, string>[]) =>
+  assertValidity(tokenValidationSuite, data, fields)
+
+export const checkTokenValidity = <T extends CombinedTokenParams>(data: T, fields?: Extract<keyof T, string>[]) =>
+  checkValidity(tokenValidationSuite, data, fields)
+
+export const extractTokenParams = ([, chainId, , givenAddress]: TokenQueryKeyType<'usdRate'>) =>
```

Comment:
Implement a factory generic function for this. It should be useful for all kind of keys. And I would call it `assertQueryKey`

Reply:
how can I extract the keys in a generic way? We only have functions for params=>key but not the other way around.